### PR TITLE
Command `sketch new` do not join sketchbook path to sketch name if only sketchname is provided

### PR DESCRIPTION
**Pull request by rsora**
This PR closes https://github.com/arduino/arduino-cli/issues/59

This PR extends the following path logic:
1. if a sketch name only is passed as parameter, assume that the sketch directory is the current directory
2. If a relative path for the sketch is passed, join it with the current directory path
3. is an absolute path for the sketch is passed, simply use it

to the `sketch new` command in addition to `compile` and `upload` commands

### CODE REVIEW

File: cli/sketch/new.go [hunk 1]
```diff
@@ -47,17 +48,21 @@ void loop() {
...
+	}
+	if err := os.MkdirAll(sketchDir, os.FileMode(0755)); err != nil {
 		feedback.Errorf("Could not create sketch directory: %v", err)
 		os.Exit(errorcodes.ErrGeneric)
 	}
-
-	sketchFile := sketchDir.Join(args[0] + ".ino")
-	if err := sketchFile.WriteFile(emptySketch); err != nil {
+	sketchName := filepath.Base(sketchDir)
+	sketchFile := filepath.Join(sketchDir, sketchName+".ino")
```
Comment (masci) [inline cli/sketch/new.go]: Not sure if worth it, but we might strip the `.ino` suffix if already present in `args[0]` (e.g. maybe an user might do `sketch new foo.ino`)

File: test/test_compile.py [hunk 2]
```diff
@@ -14,11 +14,40 @@
 # a commercial license, send an email to license@arduino.cc.
 import json
 import os
+
 import pytest
 
 from .common import running_on_ci
 
 
+def test_sketch_new(run_command):
```
Comment (masci) [inline test/test_compile.py]: we should probably create a new module `test_sketch.py` for this

File: cli/sketch/new.go (hunk 1 above)
Comment (rsora) [reply cli/sketch/new.go]: Good point, added a `TrimSuffix` upstream.

File: test/test_compile.py (hunk 2 above)
Comment (rsora) [reply test/test_compile.py]: Module created. Additionally, I applyied a small modification to the `run_command` fixture in order to have a temp working dir as a sandbox for the sketch creation e2e tests

File: test/test_new.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# This file is part of arduino-cli.
```
Comment (masci) [inline test/test_new.py]: can we name this `test_sketch.py` for consistency?

File: test/test_sketch.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+# a commercial license, send an email to license@arduino.cc.
+import os
+import platform
+
+import pytest
+
+from test.common import running_on_ci
+
+
+@pytest.mark.skipif(running_on_ci() and platform.system() == "Windows",
```
Comment (rsora) [inline test/test_sketch.py]: The e2e testing in the Github CI windows VM is temporarly disabled for the `sketch new` feature.
We are experiencing an odd behavior for windows VM in the `test` pipeline that we are not able to reproduce on a physical windows machine. The issue seems to be related to the managing of pytest temporary directories.
The test will remain disabled until a root cause is found and the fix introduced as a separate PR.

**Pull request merged**